### PR TITLE
Fix #118: Added contest page functionality

### LINKED ISSUE #118: Functionality to view all contests at one place
## **FEATURE REQUEST:**

**Description**:
As of now users struggle to search various open source competition where beginners can contribute. Competitions like Codeheat, Hactoberfest (About to start), etc. are beginner friendly but only some of them know. So if we collect all the information and list them on our website. Please let me know your opinions @OpenSourceHelpCommunity/everyone

### PR DESCRIPTION
**Pull request by abarya**
I have added the functionality to view all the contests at one page. Also added a contest form using which any user can add information regarding the upcoming contests.

It looks likes this:
image

And the form:
image

### CODE REVIEW

**Comment by abarya** [comment]
Hi @vaibhavsingh97 @tapasweni-pathak
Please review this.

**Comment by vaibhavsingh97** [comment]
@abhishekarya286 Wow!! I am impressed 👏 👏 You have done an awesome job 🎉 🎉 . Can you please fix the suggestions of Hounds CI and also can you please add `link` in the form. Also, For the visual enhancement can we have something like [this](https://getbootstrap.com/docs/3.3/components/#panels-alternatives). Keep contributing 😄

**Comment by abarya** [comment]
Hi @vaibhavsingh97,
Thanks for reviewing. I added bootstrap to the contest page and it looks fine to me now.
Also, I don't know why **The Travis CI build failed**  error. Please help me with this.

**Comment by vaibhavsingh97** [comment]
@abhishekarya286 This error came because you havn't pushed migration file. Please push migration file than this error will be resolved

**Comment by vaibhavsingh97** [comment]
@abhishekarya286 One trick : Use [autopep8](https://pypi.python.org/pypi/autopep8) tool 😄 
Also, It's best that if you have changed any thing so before migrating delete old migration file. Can you please clean unnecessary migration files 😅 .
Sorry, If you are facing too much difficulty. You can DM me anytime, I am happy to help you 😄

**Comment by abarya** [comment]
@vaibhavsingh97, thanks for your help. But still, I'm having issues with hound CI. I've tried using autopep8 but it also didn't help much.
Can you suggest something?

**Comment by vaibhavsingh97** [comment]
Hey!! @abhishekarya286 Can we talk on slack. Request your invite [here](https://opensourcehelp.herokuapp.com/) 😄

**Comment by abarya** [comment]
@vaibhavsingh97. Please review this PR.

File: oshc/main/templates/contests.html
Comment (vaibhavsingh97) [inline oshc/main/templates/contests.html]: don't use `<br>` instead make css class and use that.

File: oshc/main/views.py
Comment (vaibhavsingh97) [inline oshc/main/views.py]: Remove blank lines

File: oshc/main/views.py
Comment (vaibhavsingh97) [inline oshc/main/views.py]: Same

File: oshc/main/views.py
Comment (vaibhavsingh97) [inline oshc/main/views.py]: same

File: oshc/main/views.py
Comment (vaibhavsingh97) [inline oshc/main/views.py]: same

File: oshc/main/views.py
Comment (vaibhavsingh97) [inline oshc/main/views.py]: same

**Review by vaibhavsingh97 — CHANGES_REQUESTED** [review]
Hi!! @abhishekarya286 can you convert current UI something like this and also make sure no content should come below icon (Refer screenshot)
image

We can use AdminDateWidget to get Django admin date, this will make things simple. As of now if enter wrong fomat of date result in 500 error (this error will be reduced). After merge I will open new issue for right feedback after user enter wrong data in the fireld 😄

File: oshc/main/views.py
Comment (abarya) [reply oshc/main/views.py]: done

File: oshc/main/views.py
Comment (abarya) [reply oshc/main/views.py]: done

File: oshc/main/views.py
Comment (abarya) [reply oshc/main/views.py]: done

File: oshc/main/views.py
Comment (abarya) [reply oshc/main/views.py]: done

File: oshc/main/templates/contests.html
Comment (abarya) [reply oshc/main/templates/contests.html]: Hi @vaibhavsingh97,
you said, "also make sure no content should come below icon (Refer screenshot)"
below which icon??

**Comment by abarya** [comment]
@vaibhavsingh97,
I was unable to add a calendar using AdminDateWidget(). Other than this, please review.

File: oshc/main/migrations/0001_initial.py
Comment (tapaswenipathak) [inline oshc/main/migrations/0001_initial.py:22]: @abhishekarya286 is this pep8?

**Review by tapaswenipathak — CHANGES_REQUESTED** [review]
Thanks for the work! I have received one email per houndci comment and your commit.:smile:, thanks for patiently fixing all the issues. 

We need one more field in the table not visible on the UI which will say approved by the admin. Admin will login through django admin console and the manually say 'X' in the field for now, later we will allow admin to see submitted contest on their console. Submission of contests by the user needs to be reviewed by the admin first. After clicking on submit we need to "contest submitted for review".

File: oshc/main/migrations/0001_initial.py
Comment (abarya) [reply oshc/main/migrations/0001_initial.py:22]: Nope. I had to manually do it like that so that hound ci error doesn't come. With hound ci, you'll get something like:
`('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),`

Comment (abarya) [reply oshc/main/migrations/0001_initial.py:22]: Thanks for reviewing @tapasweni-pathak,
So what I'm trying to do now is adding up a boolean field in the contest model which will be having a  default value 0. So, when the user  will submit the contest, the admin can then review it and if he/she finds that this contest can be put up on the website then he/she can change the boolean field to 1 and the contest will be displayed on the website. Is it okay?

Comment (tapaswenipathak) [reply oshc/main/migrations/0001_initial.py:22]: Yeah, cool. Please go ahead.

**Comment by abarya** [comment]
Hi, @tapasweni-pathak,
I have added a field for approval of submissions which will be visible in the console. Also, I have added a thank you page which will be shown after submission. Kindly review.

**Comment by abarya** [comment]
@tapasweni-pathak, Any updates??

**Review by tapaswenipathak — APPROVED** [review]
@abhishekarya286 Thanks for the work!

You know I will be raising more issues after merging this, for checks, replacing form with crispy-forms but this is a great base to do the changes on! Merging this. :rocket:

**Comment by abarya** [comment]
@tapasweni-pathak, thanks for merging.

**Pull request merged**
